source/common/stats/timespan_impl.cc

```diff
@@ -0,0 +1,47 @@
+#include "common/stats/timespan_impl.h"
+
+namespace Envoy {
+namespace Stats {
+
+HistogramCompletableTimespanImpl::HistogramCompletableTimespanImpl(Histogram& histogram,
+                                                                   TimeSource& time_source)
+    : time_source_(time_source), histogram_(histogram), start_(time_source.monotonicTime()) {
+  if (!histogram.active()) {
+    return;
+  }
+
+  switch (histogram.unit()) {
+  case Histogram::Unit::Unspecified:
+  case Histogram::Unit::Bytes:
+    ASSERT(0);
```

Comment:
What is the purpose of this ASSERT? I would remove.

Edit: Sorry what is purpose of this entire switch statement?

Reply:
I think it's just to verify that this histogram has a time-based unit.  A comment to that effect would help, or maybe a helper method so it's self-documenting.

Reply:
Extracted into a method.

Correct, this is to ensure the histogram passed to the timespan is measuring time, if that's not the case, it's hard to say what `complete()` should record, how would we convert duration to say `bytes`? This is always a programming error but there's no elegant way to make this a compile time check AFAICT.